docs/www/data-transforms-tutorial.md

```diff
@@ -0,0 +1,181 @@
+
+For example, let's create a project that will change the text in your events to all uppercase:
+
+`rpk wasm generate uppercase`
+
+The project is created in a directory named for the project and including these files:
+
+```bash
+**uppercase/**
+├── package.json
+├── **src**
+│   └── main.js
+├── **test**
+│   └── main.test.js
+└── **webpack.js**
+```
+
+The most important files to know about in the project are:
+
+- /src/main.js - This file contains your transform logic and hooks into the API to define the event inputs.
+- /src/package.json - If your transform requires Node.js dependencies you need to add them to this file.
+
+## Add the transform logic
+
+Edit the `/src/main.js` file and replace the default transform logic with your JavaScript.
+The coprocessor API contains two main methods you must implement.
+The first is a method called `subscribe()` which accepts an array of 2-tuples, and the second is the transform itself called `processRecord()` that is passed a recordBatch and expects a `Map<string, Array<recordBatch>>` as the return value,
+where the key is your resultant topic name.
+
+Remember that the batch API is Bytes-In-Bytes-Out.
+We highly recommend that you build deterministic functions based on the input to make it easy to debug your applications.
+
+Just for an example, the base project already contains JavaScript to make all of the data in your events uppercase:
+
+```js
+const uppercase = (record) => {
+  const newRecord = {
+    ...record,
+    value: record.value.map((char) => {
+      if (char >= 97 && char &lt;= 122) {
+        return char - 32;
+      } else {
+        return char;
+      }
+    }),
+  };
+  return newRecord;
+}
+```
+
+If your transform requires Node.js dependencies, add them to the `/src/package.json` file.
+
+
+## Specify the source and destination topics
+
+Also in the main.js file, find these lines and edit them to specify the source and destination topics:
+
+- `transform.subscribe([["test-topic", PolicyInjection.Stored]]);` - Replace `test-topic` with the name of the topic
+    that you want to consume events from.
+	To add multiple source topics, add the topic and policy as pairs:
+    `transform.subscribe[[<topic>,<policy>],[<topic>,<policy>]]`
+- The return value of apply is `Map<string,<record_batch>>`.
+- `result.set("result", transformedRecord);` - Replace `result` with the name of the topic
+    that you want to produce the transformed events to.
+    To produce onto more than one destination topic, add another line in the format:
+    `result.set("<destination_topic>", transformedRecord);`
+
+The destination topic is created with the name format of: `<source>._<destination>_`
+
+## Prepare the script for deployment
+
+Because the transform is packaged in a Node.js project, you need to install the dependencies and build the script that runs the transform.
+
+To do this, run these commands in the project directory:
+
+```bash
+npm install
```

Comment:
@bmansheim @graphcareful Do we need to state that the version of Node used here should be at least the same version as the one we are using for the async engine?  What is that version?

Reply:
Yes, I will look into that now

Reply:
Andres recommends the Node version we use on CI, node 12

Reply:
@graphcareful So can we write: Make sure Node.js version 12 or higher is installed on the machine where you run prepare the transform bundle.